services/QuillLMS/app/controllers/skills_reports_controller.rb

```diff
@@ -0,0 +1,37 @@
+# frozen_string_literal: true
+
+class SkillsReportsController < ApplicationController
+  before_action :authorize_request, only: [:performance_breakdown_by_student]
+
+  def performance_breakdown_by_student
+    data = SkillsReport::ClassroomScoresByUser.run(
+      classroom_id:,
+      concept_id: use_sub_concept_id? ? params[:sub_concept_id].to_i : params[:concept_id].to_i,
```

Comment:
how about 
```ruby
before_action :set_ids 

private def set_ids 
  @concept_id = params[:concept_id]
  @sub_concept_id = params[:sub_concept_id]
end 
```

This would clean up the six references of `params[:[sub_]concept_id]`, making the code easier to parse

you could also do some validation checking in the before_action, returning a 4xx early, if, for example, concept_id is nil